chess_ng/board.py

```diff
@@ -134,6 +137,17 @@ def is_draw_by_repetition(
             return False
         return len(set(self.move_history[-number_of_moves:])) == number_of_teams * 2
 
+    def is_draw_by_fifty_moves(self):
+        if len(self.move_history) < 100:
+            return False
+        last_fifty_moves = self.move_history[-100:]
+        for move in last_fifty_moves:
+            # If there was a pawn move or a capture, it's a draw
+            if isinstance(move[0], Pawn) or move[2]:
+                return False
+        return True
+
+
 
 class BitBoard:
```

Comment:
BitBoard is missing the `is_draw_by_fifty_moves` method. For now a hack similar to the `BitBoard.is_draw_by_repetition` implementation can be used.

In the future, the cleaner version (avoiding duplicate code and hacks like the one I am suggesting) would be to separate the Board from any game state logic, for example by using a GameStateVisitor in the Board.move_piece method.